### ISSUE TITLE
ask(Q.real(x), Q.nonzero(x)) is True

### ISSUE THREAD

**Issue by skirpichev** [sympy/sympy#8797]
This should be None just like in the old assumptions (after #8293?).

**Comment by pelegm** [comment]
Perhaps we should have a different label for new assumptions?

**Comment by asmeurer** [comment]
I'm still unsure about what nonzero should mean.

**Comment by pelegm** [comment]
I guess it depends on other assumptions. Either way, it can't imply real. `1+I` is absolutely *nonzero*. I agree it should be `None` in this case.

**Comment by skirpichev** [comment]
> I'm still unsure about what nonzero should mean. 

@asmeurer I thought, it was discussed in #8134.  Should I add "Needs Decision"?

> I agree it should be None in this case.

@pelegm Why?  `1 + I` is nonzero.  It should be True.

**Comment by pelegm** [comment]
@skirpichev, you've misunderstood me. I agree, `1+I` is `nonzero`. It should be `True` indeed. What I meant is that `ask(Q.real(x), Q.nonzero(x))` should return `None`.

**Comment by asmeurer** [comment]
All the other "non" assumptions imply real (nonpositive, noninteger, etc.). Also if nonzero = !zero then it's basically useless, as one could always just use that.

**Comment by pelegm** [comment]
@asmeurer, you have some point there; if `nonzero` is redundant, we can either remove it, or leave it as an "alias" to `!zero`.

However, I'd take it one step further: I don't think `nonnegative` should imply real. It should imply it, assuming `number`. There are [`nonnegative` matrices](http://en.wikipedia.org/wiki/Nonnegative_matrix) as well. (same with `nonpositive`)

**Comment by pelegm** [comment]
If I understand correctly, the new assumptions do not assume `number`. So a general symbol may be a matrix as well (or a set?). Hence, if we wish to be very precise, it does not even make sense to infer `real` from `positive` or from `integer`, as there are `positive` matrices and `integer` matrices (and so on). Only when `number` is also assumed, we may infer `real`.

**Comment by skirpichev** [comment]
> All the other "non" assumptions imply real (nonpositive, noninteger, etc.).

Yes.  This is exactly one of your arguments, discussed in #8134.

Shouldn't we move on, instead of repeating same again?  I thought we all were agreed on fact modifications in #8134.

**Comment by asmeurer** [comment]
I agree it's not worthwhile to repeat the arguments without making code changes. When working on #8134 I started to second guess some of my arguments because it wasn't clear that the actual usage of the assumption matched what it seemed like it should be. So my current position on nonzero (as well as most of the real/positive/extended_real stuff) is that we should really take a look at how it is used in the code and what would change or not change with each proposed definition.
